docs/api.md

```diff
@@ -495,6 +531,61 @@ Browser websocket endpoint which can be used as an argument to
 
 You can find the `webSocketDebuggerUrl` from `http://${host}:${port}/json/version`. Learn more about the [devtools protocol](https://chromedevtools.github.io/devtools-protocol) and the [browser endpoint](https://chromedevtools.github.io/devtools-protocol/#how-do-i-access-the-browser-target).
 
+### class: BrowserContext
+
+BrowserContexts provide a way to operate multiple independent browser sessions. When browser is launched, it has
+a single BrowserContext used by default. Method `browser.newPage()` creates a page in the default browser context.
+
+If a page opens another page, e.g. with the `window.open` call, the popup will belong to the parent page browser
+context.
+
+Puppeteer allows creation of "incognito" browser contexts with `browser.createIncognitoBrowserContext()` method.
+"Incognito" browser contexts don't persist any browsing data to disk.
+
+Typical usage of a context:
+
+```js
+// Create a new incognito browser context
+const context = await browser.createIncognitoBrowserContext();
+// Create a new page inside context.
+const page = await context.newPage();
+// ... do stuff with page ...
+await page.goto('https://example.com');
+// Dispose context once its no longer needed.
+await context.dispose();
+```
+
+#### browserContext.browser()
+- returns: <[Browser]>
+
+The browser the browser context belongs to.
+
+#### browserContext.dispose()
+- returns: <[Promise]>
+
+Dispose the browser context. All the targets that belong to the browser context
+will be closed.
+
+> **NOTE** only non-incognito browser contexts can be disposed.
```

Comment:
This contradicts the example above and the note below.

Reply:
Yes.

Reply:
Yes.

Reply:
Done.